### PR TITLE
Polyfill text nodes

### PR DESCRIPTION
**Pull request by lukebatchelor**
Fix for issue #50 
Polyfills some of the functions on Text nodes to prevent issues with walk-tree in kickflip.
Added a blacklist of properties to never override on TextNodes.
Added tests for previousSibling and nextSibling when text nodes are projected somewhere different to elements

### CODE REVIEW

File: test/unit/light/polyfill.js [hunk 1]
```diff
@@ -166,4 +166,40 @@ describe('light/polyfill', function () {
       });
     });
   });
+
+  describe('when text nodes are projected to a different spot than element nodes', function () {
+    let host, light1, light2, root, namedSlot, text;
+
+    beforeEach(function () {
+      namedSlot = create('slot', {name: 'mySlot'});
+      unamedSlot = create('slot');
```
Comment (jpnelson) [inline test/unit/light/polyfill.js]: should be `unnamed`

File: src/index.js [hunk 2]
```diff
@@ -738,9 +742,17 @@ if (!('attachShadow' in document.createElement('div'))) {
     // Polyfill as much as we can and work around WebKit in other areas.
     if (canPatchNativeAccessors || polyfillAtRuntime.indexOf(memberName) === -1) {
       const nativeDescriptor = findDescriptorFor(memberName);
+      const isDefinedInTextProto = memberName in textProto;
+      const shouldOverrideInTextNode = doNotOverridePropertiesInTextNodes.indexOf(memberName) === -1;
```
Comment (jpnelson) [inline src/index.js:794]: If this was moved down to underneath Object.defineProperty, it'd make all the logic in one place, which might be more readable.

**Comment by jpnelson** [comment]
LGTM, note the comments above though :)

File: lib/index.js [hunk 3]
```diff
@@ -51,6 +51,9 @@
   // These are the protos that we need to search for native descriptors on.
   var protos = ['Node', 'Element', 'EventTarget'];
 
+  //some properties that should not be overridden in the Text prototype
+  var doNotOverridePropertiesInTextNodes = ['textContent'];
```
Comment (lukebatchelor) [inline lib/index.js:59]: Is this a bit verbose? I don't know.

Comment (treshugart) [inline lib/index.js:59]: `textNodeBlacklist` could work too, but I think this is fine.

**Comment by treshugart** [comment]
lgtm, but have to resolve your conflicts first

**Pull request merged**